pkg/kubectl/cmd/get.go

```diff
@@ -244,7 +257,12 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 			return err
 		}
 
-		if err := printer.PrintObj(obj, out); err != nil {
+                err = printer.PrintObj(obj, out)
+                if err == nil && len(args) >= 1 && strings.Contains(args[0], "pods") && cmd.Flag("show-all").Value.String() != "true" && cmd.Flag("no-headers").Value.String() != "true" {
```

Comment:
How about a helper function to do that check. Especially that you're repeating it a few times.